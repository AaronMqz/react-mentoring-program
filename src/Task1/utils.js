export const GenresList = ["ALL", "DOCUMENTARY", "COMDEY", "HORROR", "CRIME"];
