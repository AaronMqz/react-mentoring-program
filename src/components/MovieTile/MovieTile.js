import React, { useState } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faEllipsisV } from "@fortawesome/free-solid-svg-icons";
import Movie from "./MovieTile.styled";
import ContextMenu from "../ContextMenu/ContextMenu";

const MovieTile = ({ movie, handleClick }) => {
  const { image_url, movie_name, release_year, genres } = movie;
  const [itemHover, setItemHover] = useState(false);
  const [menuHover, setMenuHover] = useState(false);

  return (
    <>
      <Movie.Card
        onMouseEnter={() => setItemHover(true)}
        onMouseLeave={() => setItemHover(false)}
      >
        <Movie.Poster
          alt={movie_name}
          src={image_url}
          onClick={() => handleClick(movie)}
        />
        <Movie.MenuIcon
          hover={itemHover}
          onMouseEnter={() => setMenuHover(true)}
          onMouseLeave={() => setMenuHover(false)}
        >
          {menuHover ? (
            <ContextMenu movie={movie}></ContextMenu>
          ) : (
            <FontAwesomeIcon icon={faEllipsisV} />
          )}
        </Movie.MenuIcon>
        <Movie.Content>
          <Movie.TitleContent>
            <Movie.Title>{movie_name}</Movie.Title>
            <Movie.Year>{release_year}</Movie.Year>
          </Movie.TitleContent>
          <Movie.Tagline>{genres.toString().replace(",", " & ")}</Movie.Tagline>
        </Movie.Content>
      </Movie.Card>
    </>
  );
};

export default MovieTile;
