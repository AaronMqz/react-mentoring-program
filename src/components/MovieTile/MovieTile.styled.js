import styled from "styled-components";

const Card = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 50px;
  position: relative;
  cursor: pointer;
  transition: transform 500ms;
  &:hover {
    transform: scale(1.05);
  }
  width: 320px;
  background-color: #232323;
  padding: 10px;
`;

const Poster = styled.img`
  width: 320px;
  margin-bottom: 20px;
  min-height: 480px;
  object-fit: contain;
  background-color: #f3f3f3;
`;

const Content = styled.div`
  display: flex;
  justify-content: space-between;
  flex-direction: column;
`;

const TitleContent = styled.div`
  display: flex;
  justify-content: space-between;
  margin-bottom: 10px;
`;

const Title = styled.div`
  font-size: 20px;
  color: white;
`;

const Tagline = styled.div`
  font-size: 16px;
  color: #555555;
`;

const Year = styled.div`
  color: lightgray;
  border: solid 1px #555555;
  border-radius: 6px;
  padding: 2px 20px;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 14px;
`;

const MenuIcon = styled.span`
  display: flex;
  border-radius: 50%;
  background: #232323;
  height: 30px;
  width: 30px;
  justify-content: center;
  align-items: center;
  position: absolute;
  top: 20px;
  right: 15px;
  color: white;
  ${(props) => (props.hover ? "display:flex" : "display:none")}
`;

export default {
  Card,
  Poster,
  Content,
  TitleContent,
  Title,
  Tagline,
  Year,
  MenuIcon,
};
