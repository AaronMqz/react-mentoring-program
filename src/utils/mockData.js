export const mockData = {
  image_url: "https://m.media-amazon.com/images/I/71eHZFw+GlL._AC_SY879_.jpg",
  movie_name: "Avengers: Inifinity War",
  release_year: "2018",
  genres: ["Action", "Adventure"],
  duration: "2h 29min",
  description:
    "As the Avengers and their allies have continued to protect the world from threats too large for any one hero to handle, a new danger has emerged from the cosmic shadows: Thanos. A despot of intergalactic infamy, his goal is to collect all six Infinity Stones, artifacts of unimaginable power, and use them to inflict his twisted will on all of reality. Everything the Avengers have fought for has led up to this moment, the fate of Earth and existence has never been more uncertain.",
  rating: "8.4",
};

export const SortByOptions = [
  { id: "release_date", name: "RELEASE DATE" },
  { id: "title", name: "TITLE" },
];
