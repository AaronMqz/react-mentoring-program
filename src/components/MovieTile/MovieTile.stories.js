import MovieTile from "./MovieTile";
import { dummyMovieState } from "../../utils";

export default {
  title: "Example/MovieTile",
  component: MovieTile,
  tags: ["autodocs"],
  argTypes: { handleClick: { action: "Movie click" } },
};

export const Default = {
  args: {
    movie: dummyMovieState,
  },
};
