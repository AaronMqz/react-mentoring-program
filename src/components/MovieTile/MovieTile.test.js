import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import MovieTile from "./MovieTile";
import { movieMockData } from "../../utils";
import renderer from "react-test-renderer";

afterEach(cleanup);

describe("MovieTile Tests:", () => {
  test("1. That component is visible.", () => {
    render(<MovieTile movie={movieMockData} />);
    const movieName = screen.getByText(movieMockData.movie_name);
    expect(movieName).toBeInTheDocument();
  });

  it("2. Snapshot renders correctly", () => {
    const tree = renderer.create(<MovieTile movie={movieMockData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
