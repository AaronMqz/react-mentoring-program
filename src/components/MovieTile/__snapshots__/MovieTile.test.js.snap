// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTile Tests: 2. Snapshot renders correctly 1`] = `
<div
  className="sc-beqWaB hhCavs"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <img
    alt="Avengers: Inifinity War"
    className="sc-gueYoa cHaJKO"
    onClick={[Function]}
    src="https://m.media-amazon.com/images/I/71eHZFw+GlL._AC_SY879_.jpg"
  />
  <span
    className="sc-iAEyYk BzQWh"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-ellipsis-vertical "
      data-icon="ellipsis-vertical"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 128 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M64 360a56 56 0 1 0 0 112 56 56 0 1 0 0-112zm0-160a56 56 0 1 0 0 112 56 56 0 1 0 0-112zM120 96A56 56 0 1 0 8 96a56 56 0 1 0 112 0z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </span>
  <div
    className="sc-dmqHEX fmrsAo"
  >
    <div
      className="sc-hLseeU eAOVef"
    >
      <div
        className="sc-eDDNvR glmNaj"
      >
        Avengers: Inifinity War
      </div>
      <div
        className="sc-gLDzan fViOVN"
      >
        2018
      </div>
    </div>
    <div
      className="sc-jTrPJq GfHNu"
    >
      Action & Adventure
    </div>
  </div>
</div>
`;
